Lib/ast.py

```diff
@@ -1053,15 +1055,80 @@ def visit_AsyncWith(self, node):
         with self.block(extra=self.get_type_comment(node)):
             self.traverse(node.body)
 
+    def _str_literal_helper(
+        self, value, quote_types=("'", '"', '"""', "'''"), escape=""
+    ):
+        """Helper for writing string literals, minimising escapes.
+        Returns (possible quote types, string literal to write).
+        """
+        # Escape characters we've been told to escape and any non-printable
+        # characters. The logic for determining non-printable characters is
+        # based on that in Tools/unicode/makeunicodedata.py
+        escape = set(escape) | {
+            c for c in value
+            if c not in ' \n\t' and unicodedata.category(c)[0] in ("C", "Z")
+        }
+        qts = quote_types
```

Comment:
I don't think this is necessary, in both cases you are defining `qts = [<comprehension>]` below.

Reply:
Changed, but note we now might do the (relatively) more expensive `in` filtering on quote types we wouldn't have to.